packages/venia-concept/src/components/Checkout/form.js

```diff
@@ -1,329 +1,354 @@
-                        shippingMethod={shippingMethod}
-                        submit={this.submitShippingMethod}
-                        submitting={submitting}
-                    />
-                );
-            }
-            default: {
-                return null;
-            }
-        }
-    }
-
-    get overview() {
-        const {
-            cart,
-            classes,
-            hasPaymentMethod,
-            hasShippingAddress,
-            hasShippingMethod,
-            ready,
-            submitOrder,
-            submitting
-        } = this.props;
-
-        return (
-            <Fragment>
-                <div className={classes.body}>
-                    <Section
-                        label="Ship To"
-                        onClick={this.editAddress}
-                        showEditIcon={hasShippingAddress}
-                    >
-                        {this.shippingAddressSummary}
-                    </Section>
-                    <Section
-                        label="Pay With"
-                        onClick={this.editPaymentMethod}
-                        showEditIcon={hasPaymentMethod}
-                    >
-                        {this.paymentMethodSummary}
-                    </Section>
-                    <Section
-                        label="Use"
-                        onClick={this.editShippingMethod}
-                        showEditIcon={hasShippingMethod}
-                    >
-                        {this.shippingMethodSummary}
-                    </Section>
-                    <Section label="TOTAL">
-                        <Price
-                            currencyCode={cart.totals.quote_currency_code}
-                            value={cart.totals.subtotal || 0}
-                        />
-                        <br />
-                        <span>{cart.details.items_qty} Items</span>
-                    </Section>
-                </div>
-                <div className={classes.footer}>
-                    <Button onClick={this.dismissCheckout}>Back to Cart</Button>
-                    <SubmitButton
-                        submitting={submitting}
-                        valid={ready}
-                        submitOrder={submitOrder}
-                    />
-                </div>
-            </Fragment>
-        );
-    }
+const isSubmitDisabled = (busy, valid) => busy || !valid;
+
+const EditableForm = props => {
+    const {
+        editing,
+        editOrder,
+        submitShippingAddress,
+        submitShippingMethod,
+        submitPaymentMethodAndBillingAddress,
+        submitting,
+        isAddressIncorrect,
+        incorrectAddressMessage,
+        directory: { countries }
+    } = props;
+
+    const handleCancel = useCallback(() => {
+        editOrder(null);
+    }, [editOrder]);
+
+    const handleSubmitAddressForm = useCallback(
+        formValues => {
+            submitShippingAddress({
+                type: 'shippingAddress',
```

Comment:
I don't think we need to forward these `type` props since we have specific functions for submission of each type. Maybe a remnant from some earlier code? I see two ways to handle this, and I'd like opinions:

1) Remove `type`. Easiest "fix".
2) Refactor into a single `submitForm` action that accepts a `type` and handles logic internally instead of having multiple different `submit` functions. Benefit of only having to thread one function prop down instead of three.

Reply:
I opted for option 1 right now since nothing is using `type`. And we can easily refactor the actions later.